traitsui/testing/tester/registry.py

```diff
@@ -0,0 +1,273 @@
+
+        registry = TargetRegistry()
+        registry.register_handler(
+            target_class=traitsui.qt4.button_editor.SimpleEditor,
+            interaction_class=traitsui.testing.tester.command.MouseClick,
+            handler=mouse_click_qt_button,
+        )
+
+    Similarly, a wx implementation of clicking a button can be registered
+    to the registry (the content of ``mouse_click_wx_button`` is not shown)::
+
+        registry.register_handler(
+            target_class=traitsui.wx.button_editor.SimpleEditor,
+            interaction_class=traitsui.testing.tester.command.MouseClick,
+            handler=mouse_click_wx_button,
+        )
+
+    Then this registry can be used with the ``UITester`` and ``UIWrapper`` to
+    support ``UIWrapper.perform`` and ``UIWrapper.inspect``.
+
+    Registering location solver (register_solver)
+    ---------------------------------------------
+
+    Resoling a location on a UI target is logically similar to making a query
+    for a nested UI target. This query is separated out to support the
+    ``UIWrapper.locate`` method independently of the query method
+    ``UIWrapper.inspect``.
+
+    The locator type can be any subclass of ``type``. There are predefined
+    locators in ``traitsui.testing.tester.locator``.
+
+    For example, suppose a UI target has a nested UI and a button. Both the
+    nested UI and the button are UI targets that can be located in the
+    container target.
+
+    Suppose we have a locator type for the nested UI::
+
+        class NestedUI:
+            pass
+
+    And there is an enum for the widget type::
+
+        class WidgetType(Enum):
+            button = "button"
+
+    The solvers for these locators may look like this::
+
+        def get_nested_ui(wrapper, _):
+            return wrapper.target._nested_ui
+
+        def get_widget_by_type(wrapper, location):
+            if location == WidgetType.button:
+                return wrapper.target._button
+            raise ValueError("Other widget type not supported")
+
+    The first argument is an instance of ``UIWrapper`` that wraps the container
+    UI target, in this case, the UI target that holds a nested UI and a button.
+
+    The second argument is an instance of the locator type.
+
+    The solvers can then be registered for the container UI target::
+
+        registry = LocationRegistry()
+        registry.register_solver(
+            target_class=MyUIContainer,
+            locator_class=NestedUI,
+            solver=get_nested_ui,
+        )
+        registry.register_solver(
+            target_class=MyUIContainer,
+            locator_class=WidgetType,
+            solver=get_widget_by_type,
+        )
+
+    Then this registry can be used with the ``UITester`` and ``UIWrapper`` to
+    support ``UIWrapper.locate``. If the nested UI has other locator solvers,
+    then it would be possible to do chain location resolutions, like this::
+
+        container.locate(NestedUI()).locate(Index(1))
```

Comment:
Is Index defined in this PR?

Reply:
No it isn't, I will change the `Index(1)` to `NestedUI()` which is added here.